ignite/engine/__init__.py

```diff
@@ -41,6 +47,208 @@ def _prepare_batch(
     )
 
 
+def _check_arg(on_tpu: bool, amp_mode: Optional[str], scaler: Optional["torch.cuda.amp.GradScaler"]) -> Optional[str]:
+    """Checking tpu, amp and GradScaler instance combinations."""
+    if on_tpu and not idist.has_xla_support:
+        raise RuntimeError("In order to run on TPU, please install PyTorch XLA")
+
+    if amp_mode and on_tpu:
+        raise ValueError("amp_mode cannot be used with xla device. Consider using amp_mode=None or device='cuda'.")
+
+    if scaler is not None and not amp_mode:
+        warn("scaler argument is provided, but amp_mode is not. Consider using amp_mode='amp'.")
+
+    return "tpu" if on_tpu else amp_mode
+
+
+def supervised_trainer_step(
+    model: torch.nn.Module,
+    optimizer: torch.optim.Optimizer,
+    loss_fn: Union[Callable, torch.nn.Module],
+    device: Optional[Union[str, torch.device]],
+    non_blocking: bool,
+    prepare_batch: Callable,
+    output_transform: Callable,
+) -> Callable:
+    """Helper function defined the training step.
```

Comment:
Update the docstring wrt default values.

Reply:
The docstring must be updated, see docstring of `create_supervised_trainer`. Thanks!